### PR TITLE
Expose `Basis::set_orthogonal_index` as a GridMap method

### PR DESCRIPTION
**Pull request by rafallus**
This can be useful to replicate the transform of a GridMap cell by retrieving the index with the `get_cell_item_orientation` method.

### CODE REVIEW

**Review by fire — APPROVED** [review]
Approving

1. Assists with an existing Godot engine area: gridmaps.
1. The function already exists in c++.
1. I can't justify recreating the code by force in gdscript.

Are there any other functions like this?

**Comment by akien-mga** [comment]
It seems that both `Basis::get_orthogonal_index` (already exposed) and `Basis::set_orthogonal_index` are both used only in the GridMap editor and very specific to GridMap.

I feel like they should likely be moved to be helper methods in the GridMap module instead of being member methods of `Basis` which are not relevant in the core API.

Something like:
```
Basis GridMapEditor::get_basis_with_orthogonal_index(int p_index);
int GridMapEditor::get_orthogonal_index_from_basis(const Basis &p_basis);
```

**Review by fire — DISMISSED** [review]
Changing the review to request changes as noted by remi.

**Comment by rafallus** [comment]
> It seems that both `Basis::get_orthogonal_index` (already exposed) and `Basis::set_orthogonal_index` are both used only in the GridMap editor and very specific to GridMap.
> 
> I feel like they should likely be moved to be helper methods in the GridMap module instead of being member methods of `Basis` which are not relevant in the core API.
> 
> Something like:
> 
> ```c++
> Basis GridMapEditor::get_basis_with_orthogonal_index(int p_index);
> int GridMapEditor::get_orthogonal_index_from_basis(const Basis &p_basis);
> ```

Both functions are now in GridMap. I also added a convenience function to get the basis of a cell.

File: modules/gridmap/grid_map.cpp [hunk 1]
```diff
@@ -409,6 +409,26 @@ int GridMap::get_cell_item_orientation(const Vector3i &p_position) const {
...
+	}
+
+	return get_basis_with_orthogonal_index(orientation);
+}
+
+Basis GridMap::get_basis_with_orthogonal_index(int p_index) const {
+	Basis basis;
+	basis.set_orthogonal_index(p_index);
+	return basis;
+}
```
Comment (akien-mga) [inline modules/gridmap/grid_map.cpp:426]: I meant that the whole `Basis::set_orthogonal_index` should be moved to GridMap, as it's not needed to be in `Basis` itself.

**Review by akien-mga — APPROVED** [review]

**Review by reduz — APPROVED** [review]

**Comment by aaronfranke** [comment]
@rafallus This was discussed in a PR meeting. This looks good, but it needs to be rebased on the latest master.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**